src/Analyzers/Orchestration/DateTimeOrchestrationAnalyzer.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+using Microsoft.CodeAnalysis;
+using Microsoft.DurableTask.Analyzers.Helpers;
+
+namespace Microsoft.DurableTask.Analyzers.Orchestration;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class DateTimeOrchestrationAnalyzer : OrchestrationAnalyzer
+{
+    public const string DiagnosticId = "DURABLE0001";
+
+    static readonly DiagnosticDescriptor Rule = new(
+        DiagnosticId,
+        "DateTime calls must be deterministic inside an orchestration function",
```

Comment:
Do we have plans to add these strings to a resx?

Reply:
I think that we should add them to a resx file like in the In Proc Analyzer - https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/WebJobs.Extensions.DurableTask.Analyzers/Resources.resx

Reply:
Moved strings to resources file ✅